src/hats_import/catalog/run_import.py

```diff
@@ -148,3 +154,23 @@ def run(args, client):
             step_progress.update(1)
             assert is_valid_catalog(args.catalog_path)
             step_progress.update(1)
+
+
+def create_data_thumbnail(args, client, resume_plan):
+    """Create a thumbnail of the data in the catalog, top row from each partition."""
+    output_catalog_path = get_upath(args.catalog_path)
+    dataset_subdir = output_catalog_path / "dataset"
+    _, dataset = io.read_parquet_dataset(
+        dataset_subdir,
+        ignore_prefixes=["intermediate", "_common_metadata", "_metadata"],
+        exclude_invalid_files=True,
+    )
+    futures = [
+        client.submit(lambda pq_path: pd.read_parquet(pq_path, engine="pyarrow").head(1), p)
+        for p in dataset.files
+    ]
+    resume_plan.wait_for_thumbnail(futures)
+    if futures:
+        thumbnail = pd.concat(client.gather(futures), ignore_index=True)
+        thumbnail_file = output_catalog_path / "data_thumbnail.parquet"
+        thumbnail.to_parquet(thumbnail_file, index=False)
```

Comment:
This might not work with non-local file systems. Pandas will not unpack the filesystem from the UPath.

Reply:
pq.write_table(thumbnail, thumbnail_file.path, filesystem=thumbnail_file.fs)